invites.go

```diff
@@ -106,11 +106,38 @@ func (srv *invitesAPI) ListInvites(ctx context.Context, in *accountsv1.ListInvit
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	_, err = srv.authenticate(ctx)
+	token, err := srv.authenticate(ctx)
 	if err != nil {
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
+	// If you try to fetch notes from a specific group
+	if in.GroupId != "" {
+		m, err := srv.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: in.GroupId, AccountId: token.UserID.String()})
+
+		// If you are already in the group and not an Admin, you should ask only for invites where you are the sender
+		if err == nil && m.Member.Role != auth.RoleAdmin && in.SenderAccountId != token.UserID.String() {
+			return nil, status.Error(codes.Unauthenticated, "as a user, you can only list your invites")
+		} else if err != nil {
+			e, ok := status.FromError(err)
+			if !ok {
+				return nil, status.Error(codes.Internal, e.Message())
+			}
+			if e.Code() != codes.NotFound {
+				return nil, err
+			}
+
+			// If you are not in the current group, you should ask only for invites where you are the recipient
+			if in.RecipientAccountId != token.UserID.String() {
+				return nil, status.Error(codes.Unauthenticated, "you can only fetch your invites for this group")
+			}
+		}
+
+		// If you are not trying to fetch invites from a group, you should only fetch those where you are sender or recipient
+	} else if in.RecipientAccountId != token.UserID.String() && in.SenderAccountId != token.UserID.String() {
+		return nil, status.Error(codes.Unauthenticated, "you have no rights to fetch other users invites")
+	}
+
 	if in.Limit == 0 {
 		in.Limit = 20
 	}
```

Comment:
Line 127/154 you say 

> // NOTE: As every account are apparently accessible to list and get from every account as long as you are authentificated, I did the same with invites

Is that still d'actualité?

Reply:
Same thing, maybe it's not a problem, but a bit further down in the endpoint you do
```go
	invites, err := srv.inviteRepo.List(ctx, &models.ManyInvitesFilter{
		SenderAccountID:    &in.SenderAccountId, // <--
		RecipientAccountID: &in.RecipientAccountId, // <--
		GroupID:            &in.GroupId, // <--
	}, &models.Pagination{Offset: int64(in.Offset), Limit: int64(in.Limit)})
	if err != nil {
		return nil, statusFromModelError(err)
	}
```
Shouldn't you be passing nil if the user has put no filter on sender, recipient or group? Does it not mess with the `models` layer? Just curious

Reply:
For first comment it is not anymore !